### ISSUE TITLE
Create Class Interface, etc.

### ISSUE THREAD

**Issue by PedroAvila** [microsoft/vscode-dotnettools#682]
Type: <b>Bug</b>

It does not allow me to create files such as classes, interfaces etc. This is the error I get.
error ExitCode(2097158) Details No template found matching: 'class' To display the list of installed templates, run: dotnet new --list To search for templates in NuGet.org, run: dotnet new class --search . It is probably caused by the extension contributing to csdevkit.addNewFile.

Extension version: 1.0.14
VS Code version: Code 1.84.0 (d037ac076cee195194f93ce6fe2bdfe2969cc82d, 2023-11-01T11:29:04.398Z)
OS version: Windows_NT x64 10.0.22621
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|11th Gen Intel(R) Core(TM) i7-11700 @ 2.50GHz (16 x 2496)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: enabled_on<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled|
|Load (avg)|undefined|
|Memory (System)|15.83GB (3.47GB free)|
|Process Argv|--crash-reporter-id 9630f65d-16d3-48a9-b724-3df94a0ff1af|
|Screen Reader|no|
|VM|0%|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vsreu685:30147344
python383cf:30185419
vspor879:30202332
vspor708:30202333
vspor363:30204092
vslsvsres303:30308271
vserr242cf:30382550
pythontb:30283811
vsjup518:30340749
pythonptprofiler:30281270
vsdfh931cf:30280410
vshan820:30294714
vstes263cf:30335440
vscod805cf:30301675
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
vsaa593cf:30376535
pythonvs932:30410667
py29gd2263:30880072
vsclangdf:30486550
c4g48928:30535728
dsvsc012cf:30540253
pynewext54:30695312
azure-dev_surveyone:30548225
282f8724:30602487
f6dab269:30613381
showlangstatbar:30737416
pythonfmttext:30731395
fixshowwlkth:30771522
showindicator:30805244
pythongtdpath:30769146
i26e3531:30792625
pythonnosmt12:30797651
pythonidxpt:30866567
pythonnoceb:30805159
synctok:30869157
dsvsc013:30795093
dsvsc014:30804076
dsvsc015:30845448
pythontestfixt:30871694
pythonregdiag2:30871582
pythonmypyd1:30879173
pythoncet0:30874138
pythontbext0:30879054

```

</details>

**Comment by smitpatel** [comment]
@PedroAvila - Can you share which version of SDK are you using?

**Comment by smitpatel** [comment]
Similar #674 

I have identified root cause here. We detect missing template based on exit code returned by `dotnet new` command. The exit codes for the command has changed from SDK version 6.0 to 7.0. Our current code in C# dev kit utilizes codes which are in SDK 7.0. We should also add exit code from 6.0 in our detection logic. I verified that on a windows machine exit code 2097158 refers to template not found.
